### PR TITLE
docs: Add clarity on reverse proxies and supported versions to troubleshooting section

### PR DESCRIPTION
**Pull request by webvictim**
As noted in https://github.com/gravitational/teleport/issues/32747#issuecomment-2058677726, the docs aren't super clear as to whether nginx and other reverse proxies are supported or not. Now that every supported Teleport version (13.0 and up) supports reverse proxies, I've updated the language to make it clearer.

We can eventually remove the "prior to version 13.0" section, but I wanted to leave it for now as many folks aren't familiar with choosing older versions of the docs from the dropdown and I don't want them to be confused.

### CODE REVIEW

**Review by ptgott — APPROVED** [review]

File: docs/pages/includes/tls-multiplexing-warnings.mdx [hunk 1]
```diff
@@ -1,17 +1,35 @@
...
+    nginx.ingress.kubernetes.io/backend-protocol: "HTTPS"
+```
+
+Deploying Teleport behind Cloudflare, whether using its proxy ("orange-clouding") or tunnels
+(`cloudflared`) should work with Teleport version 15.1 or higher. See the [TLS Routing FAQ]
+(../../architecture/tls-routing/#will-tls-routing-work-behind-my-reverse-proxy) for more details.
+
+#### Prior to Teleport version 13.0
+
+Prior to Teleport version `13.0`, using Teleport's TLS routing mode behind a layer 7 (HTTP/HTTPS) proxy is not supported,
```
Comment (zmb3) [inline docs/pages/includes/tls-multiplexing-warnings.mdx:25]:
```suggestion
Prior to Teleport version 13.0, using Teleport's TLS routing mode behind a layer 7 (HTTP/HTTPS) proxy is not supported,
```

No need for the backticks here. Let's just refer to a version number like any other text (and just like you did with 15.1 on Line 20)

File: docs/pages/includes/tls-multiplexing-warnings.mdx [hunk 2]
```diff
@@ -1,17 +1,35 @@
...
+due to these proxies terminating TLS themselves and then rewriting their requests to the upstream service, stripping
+the additional SNI/ALPN parts of the request in the process.
+
 For older versions, in order for ALPN to work correctly, the Teleport Proxy
 Service must terminate TLS itself.
 
-Broadly, this means that Teleport's TLS routing functionality is incompatible with:
+Broadly, this means that prior to Teleport version `13.0`, Teleport's TLS routing functionality is incompatible with:
```
Comment (webvictim) [inline docs/pages/includes/tls-multiplexing-warnings.mdx:32]:
```suggestion
Broadly, this means that prior to Teleport version 13.0, Teleport's TLS routing functionality is incompatible with:
```

File: docs/pages/includes/tls-multiplexing-warnings.mdx [hunk 3]
```diff
@@ -25,10 +43,10 @@ of a correctly configured Teleport cluster.
 If in doubt, remove all load balancers/proxies from the equation and connect Teleport clients or agent processes directly to
 Teleport's web port to isolate the issue.
 
-To use Teleport behind a reverse proxy, you should either:
+To use Teleport behind a reverse proxy prior to Teleport version `13.0`, you should either:
```
Comment (webvictim) [inline docs/pages/includes/tls-multiplexing-warnings.mdx:46]:
```suggestion
To use Teleport behind a reverse proxy prior to Teleport version 13.0, you should either:
```

File: docs/pages/includes/tls-multiplexing-warnings.mdx [hunk 4]
```diff
@@ -1,17 +1,35 @@
-Prior to Teleport `13.0`, using Teleport's TLS routing mode behind a layer 7 (HTTP/HTTPS) proxy is generally not supported, due to
-these proxies terminating TLS themselves and then rewriting their requests to the upstream service, stripping
-the additional SNI/ALPN parts of the request in the process.
+#### Teleport version 13.0+
 
 Support for TLS routing behind layer 7 (HTTP/HTTPS) load balancers and reverse
 proxies is available starting from Teleport `13.0`. Please ensure your Teleport
```
Comment (webvictim) [inline docs/pages/includes/tls-multiplexing-warnings.mdx:4]:
```suggestion
proxies is available starting from Teleport 13.0. Please ensure your Teleport
```

File: docs/pages/includes/tls-multiplexing-warnings.mdx [hunk 5]
```diff
@@ -1,17 +1,35 @@
...
+
+```yaml
+annotations:
+  ingress:
+    nginx.ingress.kubernetes.io/backend-protocol: "HTTPS"
+```
+
+Deploying Teleport behind Cloudflare, whether using its proxy ("orange-clouding") or tunnels
+(`cloudflared`) should work with Teleport version 15.1 or higher. See the [TLS Routing FAQ]
+(../../architecture/tls-routing/#will-tls-routing-work-behind-my-reverse-proxy) for more details.
```
Comment (webvictim) [inline docs/pages/includes/tls-multiplexing-warnings.mdx:21]:
```suggestion
(`cloudflared`) should work with Teleport version 15.1 or higher. See the
[TLS Routing FAQ](../../architecture/tls-routing/#will-tls-routing-work-behind-my-reverse-proxy) for more details.
```

**Review by greedy52 — APPROVED** [review]

File: docs/pages/includes/tls-multiplexing-warnings.mdx [hunk 6]
```diff
@@ -1,17 +1,35 @@
...
+
+```yaml
+annotations:
+  ingress:
+    nginx.ingress.kubernetes.io/backend-protocol: "HTTPS"
+```
+
+Deploying Teleport behind Cloudflare, whether using its proxy ("orange-clouding") or tunnels
+(`cloudflared`) should work with Teleport version 15.1 or higher. See the
+[TLS Routing FAQ](../../architecture/tls-routing/#will-tls-routing-work-behind-my-reverse-proxy) for more details.
```
Comment (webvictim) [inline docs/pages/includes/tls-multiplexing-warnings.mdx:21]:
```suggestion
[TLS Routing FAQ](../../architecture/tls-routing.mdx#will-tls-routing-work-behind-my-reverse-proxy) for more details.
```

File: docs/pages/includes/tls-multiplexing-warnings.mdx [hunk 7]
```diff
@@ -1,17 +1,35 @@
...
+
+```yaml
+annotations:
+  ingress:
+    nginx.ingress.kubernetes.io/backend-protocol: "HTTPS"
+```
+
+Deploying Teleport behind Cloudflare, whether using its proxy ("orange-clouding") or tunnels
+(`cloudflared`) should work with Teleport version 15.1 or higher. See the
+[TLS Routing FAQ](../../architecture/tls-routing.mdx#will-tls-routing-work-behind-my-reverse-proxy) for more details.
```
Comment (webvictim) [inline docs/pages/includes/tls-multiplexing-warnings.mdx:21]:
```suggestion
[TLS Routing FAQ](../architecture/tls-routing.mdx#will-tls-routing-work-behind-my-reverse-proxy) for more details.
```

Comment (webvictim) [reply docs/pages/includes/tls-multiplexing-warnings.mdx:21]: Trying this to fix the linter, it seems confused.

**Pull request merged**